### DISCUSSION TITLE
Set up using HSM

### DISCUSSION THREAD

**Discussion by mahendrahendi** [xipki/xipki#240] [Q&A] [answered]
Hi

Currently i'm studying xipki and trying to using HSM

i already follow the step ,  what i've done is
1. specify the lib from my hsm inside xipki/security/pkcs11.json
2. set password inside xipki/security/pkcs11.json

when i try to setup CA i got this error  "Error executing command: login failed as user CKU_USER: CKR_PIN_INCORRECT"

also i have no idea with this command preload <start script>, i just try something like this , preload -c <my ocs name> ./karaf

do i miss something?

**update :**

seems work after i use preload to start tomcat

got new error when setup a CA , i ran this command `source `/xipki/ca-setup/cacert-none/setup-rsa-p11.script ``

and this is the error showed` Error executing command: could not refresh token for signer type pkcs11: could not refreshed modules`

**Comment by xipki** [comment, marked as answer]
There are two pkcs11.json to configure. The one is in tomcat folder, which will be used by the CA server. The other one is in the xipki-cli, which will be used by the xipki-cli. You need to configure both (you can simple configure one and copy it).

And you need to start both tomcat (bin/start.sh) and xipki-cli (bin/karaf) with preload.

**Reply by mahendrahendi** [reply]
Thanks Lijun Liao

**Comment by mahendrahendi** [comment]
Hi @xipki 

can i have a question here

i have my own tool to  generate CA key and certificate in HSM, my tool is save the key inside HSM and certificate on local storage
now i want to import my OCSP certificate using xipki CLI's  "update-cert-p11"

it say "Error executing command: keyId may not be null" when import the certificate
what i missing here?

actually i'm going  to use OCSP responder along with my own certificate and key that generate by our CA tool
it seem failed when i specify key/cert label to my own genereted key/cert, but no problem with xipki key/cert 

also may i know what is the difference between add-cert-p11 and update-cert-p11

what i missing here?

thanks
